f8a_worker/storages/s3.py

```diff
@@ -154,13 +154,19 @@ def store_file(self, file_path, object_key):
         :param object_key: object key under which the file should be stored
         :return: object version or None if versioning is off
         """
-        with open(file_path, 'rb') as f:
-            return self.store_blob(f.read(), object_key)
+
+        f = None
+        try:
+            f = open(file_path, 'rb')
+            return self.store_blob(f, object_key)
+        finally:
+            if f is not None:
+                f.close()
 
     def store_blob(self, blob, object_key):
         """ Store blob on S3
 
-        :param blob: bytes to be stored
+        :param blob: bytes or stream to be stored
```

Comment:
[S3.Object.put() doc](http://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Object.put) says it's `bytes or seekable file-like object`, which I like more, but I guess this is OK too.